### PR TITLE
Feature/api refactoring

### PR DESCRIPTION
**Pull request by apfelfabrik**
Moves current REST API as well as the timelines API into their own versions of one application API. Core becomes v1, timelines becomes v2. There is no combined version 3 yet.
- [x] Merge from timelines repo
- [x] Get the tests to run
- [x] Restore original migrations
- [x] Create new migration
- [x] FIX ALL THE TESTS!
- [x] Merge feature/rails3

This one is work in progress and does neither pass specs nor merge with the core flawlessly. Proceed with caution!

### CODE REVIEW

**Comment by apfelfabrik** [comment]
Specs and cukes run again, with 1/3 of specs green \o/

**Comment by apfelfabrik** [comment]
Rebased and squashed a bit to remove a bad feature/rails3 merge. Needs to be merged again, and final migration is still not done.

**Comment by apfelfabrik** [comment]
1229 examples, 289 failures, 4 pending

**Comment by apfelfabrik** [comment]
1229 examples, 199 failures, 4 pending

**Comment by apfelfabrik** [comment]
1229 examples, 7 failures, 4 pending

**Comment by apfelfabrik** [comment]
1234 examples, 0 failures, 4 pending

File: app/helpers/pagination_helper.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+           elsif options[:offset] && options[:limit]
+
+             begin
+              # + 1 as page is not 0 but 1 based
+              options[:offset].to_i/per_page_param(options) + 1
+             rescue ZeroDivisionError
+               1
+             end
+
```
Comment (hschink) [inline app/helpers/pagination_helper.rb:72]: Is their a purpose for those blank lines.

**Pull request merged**